### ISSUE TITLE
Calling translation with plural is returning an array

### ISSUE THREAD

**Issue by Yovach** [QuiiBz/next-international#147]
Hello,
The `t` function returned by `getI18n` should return a string instead of an array on server actions (or non-RSC)

eg:

```typescript

"use server";

import { getI18n } from "#locales/server";

export async function hello(name: string) {
  const t = await getI18n();
  return {
    message: t("welcome", { name, })
  };
}

```

Because of the param 'name', value of 'message' is an array.

Is there a way to make getI18n take a parameter like "forceFlat" or something like that?
At the moment, I have to make a function to convert the result of `t` into a string because the typing tells me "string" but may return a list (probably a ReactNode).

Thanks!

**Comment by QuiiBz** [comment]
`t()` always returns a `string`, unless a parameter is some kind of `ReactNode`. In this case, it will return a `ReactNode` because it has to somehow concatenate the translation string with the React elements given in the parameters.

![Screenshot 2023-09-02 at 14 14 17](https://github.com/QuiiBz/next-international/assets/43268759/0b306320-6a88-48e1-904e-330894dc89ad)

![Screenshot 2023-09-02 at 14 15 54](https://github.com/QuiiBz/next-international/assets/43268759/c602c404-2faf-4412-b1c9-8bc260742685)

> I have to make a function to convert the result of t into a string because the typing tells me "string" but may return a list

You shouldn't need to do this based on my explanation above. I haven't tried to use next-international with Server Actions since it's experimental. Do you mean that it currently doesn't work?

**Comment by QuiiBz** [comment]
I was curious and tried using the following code to return a translated message by giving a parameter from the client to the server action:

![Screenshot 2023-09-02 at 14 28 55](https://github.com/QuiiBz/next-international/assets/43268759/70366df2-08a9-457d-a04e-f20bd737d14c)

It seems to work as expected:

![Screenshot 2023-09-02 at 14 29 45](https://github.com/QuiiBz/next-international/assets/43268759/5f7a1a90-f921-44a5-90ae-a43ba248ec71)

**Comment by Yovach** [comment]
Oh sorry, my example was different from the one where I had the problem, I didn't test the one I shared because I thought it would mess up too.


Here's a case where the issue is present and is a behavior not necessarily related to "server actions" : 

```typescript
// locales/en.ts
export default {
  "time.minutes#one": "1 minute",
  "time.minutes#other": "{count} minutes",

  "please_wait": "Please wait {time}",
} as const;
```

```typescript
// `page.tsx`
export default async function Page() {
  const t = await getI18n();

  // it returns "string" when it is an array (or a ReactNode) in the console
  const wrongType = t("please_wait", { time: t("time.minutes", { count: 5 }) });

  // it returns "string" which is correct
  const validType = t("please_wait", { time: "5 minutes" });
  console.log({ wrong: wrongType, valid: validType });

  return (
    <>
      <span>{wrongType}</span>
      <span>{validType}</span>
    </>
  );
}
```

**Comment by QuiiBz** [comment]
Thanks for the precision. The issue doesn't seem to be caused by plurals, but by parameters that aren't `string` (in your case, `5` is a `number`):

https://github.com/QuiiBz/next-international/blob/040728a29fee89869e4cf9b2699f30a27e76741a/packages/next-international/src/common/create-t.ts#L78

This is technically a breaking change and will be included in the v1 release (see #126) which will be out soon.

**Comment by Yovach** [comment]
Thank you for the reply!

**Comment by QuiiBz** [comment]
I've just released [1.0.0-rc.3](https://www.npmjs.com/package/next-international/v/1.0.0-rc.3) that includes this fix, but I'll need to read the release note in the description of #126 if you want to upgrade.
